src/IB/IBHydrodynamicSurfaceForceEvaluator.cpp

```diff
@@ -428,55 +559,58 @@ IBHydrodynamicSurfaceForceEvaluator::fillPatchData(Pointer<PatchHierarchy<NDIM>
-        if (p_ins_bc_coef)
-        {
-            p_ins_bc_coef->setTargetVelocityPatchDataIndex(d_u_idx);
-            InterpolationTransactionComponent p_transaction_comp(d_p_idx,
-                                                                 p_current_idx,
-                                                                 /*DATA_REFINE_TYPE*/ "CONSERVATIVE_LINEAR_REFINE",
-                                                                 /*USE_CF_INTERPOLATION*/ true,
-                                                                 /*DATA_COARSEN_TYPE*/ "CUBIC_COARSEN",
-                                                                 /*BDRY_EXTRAP_TYPE*/ "LINEAR",
-                                                                 /*CONSISTENT_TYPE_2_BDRY*/ false,
-                                                                 p_ins_bc_coef,
-                                                                 Pointer<VariableFillPattern<NDIM> >(nullptr));
-            Pointer<HierarchyGhostCellInterpolation> hier_p_bdry_fill = new HierarchyGhostCellInterpolation();
-            hier_p_bdry_fill->initializeOperatorState(p_transaction_comp, patch_hierarchy);
-            hier_p_bdry_fill->setHomogeneousBc(false);
-            hier_p_bdry_fill->fillData(fill_time);
-        }
-        else if (p_vc_ins_bc_coef)
-        {
-            p_vc_ins_bc_coef->setTargetVelocityPatchDataIndex(d_u_idx);
-            InterpolationTransactionComponent p_transaction_comp(d_p_idx,
-                                                                 p_current_idx,
-                                                                 /*DATA_REFINE_TYPE*/ "CONSERVATIVE_LINEAR_REFINE",
-                                                                 /*USE_CF_INTERPOLATION*/ true,
-                                                                 /*DATA_COARSEN_TYPE*/ "CUBIC_COARSEN",
-                                                                 /*BDRY_EXTRAP_TYPE*/ "LINEAR",
-                                                                 /*CONSISTENT_TYPE_2_BDRY*/ false,
-                                                                 p_vc_ins_bc_coef,
-                                                                 Pointer<VariableFillPattern<NDIM> >(nullptr));
-
-            Pointer<HierarchyGhostCellInterpolation> hier_p_bdry_fill = new HierarchyGhostCellInterpolation();
-            hier_p_bdry_fill->initializeOperatorState(p_transaction_comp, patch_hierarchy);
-            hier_p_bdry_fill->setHomogeneousBc(false);
-            hier_p_bdry_fill->fillData(fill_time);
-        }
-        else
-        {
-            TBOX_ERROR(d_object_name << "::IBHydrodynamicSurfaceForceEvaluator():\n"
-                                     << " no valid pressure boundary condition object registered with INS integrator.\n"
-                                     << " This statement should not have been reached");
-        }
+    // Fill ghost cells for pressure
```

Comment:
Note that we need to do this regardless of whether nor not the density is constant, so this fixes a bug.